app/src/main/java/com/mapbox/maps/testapp/examples/LargeGeojsonPerformanceActivity.kt

```diff
@@ -0,0 +1,167 @@
+import com.mapbox.maps.extension.style.layers.generated.symbolLayer
+import com.mapbox.maps.extension.style.layers.properties.generated.IconAnchor
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.plugin.animation.MapAnimationOptions
+import com.mapbox.maps.plugin.animation.flyTo
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import kotlinx.android.synthetic.main.activity_large_geojson.*
+
+/**
+ * Example of passing large geojson to verify it does not block UI thread.
+ */
+class LargeGeojsonPerformanceActivity : AppCompatActivity() {
+
+  private lateinit var routePoints: FeatureCollection
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_large_geojson)
+
+    routePoints = FeatureCollection.fromFeature(
+      Feature.fromGeometry(
+        LineString.fromPolyline(
+          DirectionsResponse.fromJson(
+            AnnotationUtils.loadStringFromAssets(
+              this@LargeGeojsonPerformanceActivity, "long_route.json"
+            )
+          ).routes()[0].geometry()!!,
+          6
+        )
+      )
+    )
+
+    mapView.getMapboxMap()
+      .apply {
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(LONGITUDE, LATITUDE))
+            .zoom(START_ZOOM)
+            .build()
+        )
+        flyTo(
+          CameraOptions.Builder()
+            .zoom(END_ZOOM)
+            .build(),
+          MapAnimationOptions.mapAnimationOptions {
+            duration(10_000L)
+          }
+        )
+        loadStyle(
+          style(Style.DARK) {
+            +geoJsonSource("source") {
+              featureCollection(routePoints)
+            }
+            +geoJsonSource("source2") {
+              featureCollection(routePoints)
+            }
+            +geoJsonSource("source3") {
+              featureCollection(routePoints)
+            }
+            +lineLayer("layer", "source") {
+              lineColor("blue")
+            }
+            +lineLayer("layer2", "source2") {
+              lineColor("yellow")
+              lineOffset(5.0)
+            }
+            +lineLayer("layer3", "source3") {
+              lineColor("red")
+              lineOffset(10.0)
+            }
+            +image("icon") {
+              bitmap(BitmapFactory.decodeResource(resources, R.drawable.blue_marker_view))
+            }
+            +geoJsonSource("source_marker") {
+              geometry(Point.fromLngLat(LONGITUDE, LATITUDE))
+            }
+            +symbolLayer("layer_marker", "source_marker") {
+              iconImage("icon")
+              iconAnchor(IconAnchor.BOTTOM)
+            }
+          }
+        ) {
+          loadAdditionalGeoJsonAfter(it)
+        }
+      }
```

Comment:
nit duplicate values, could be replaced with a constant

Reply:
Example will be revisited for sure, thanks for heads up!